### ISSUE TITLE
No audio when playing two files after upgrading libmpv

### ISSUE THREAD

**Issue by low-batt** [mpv-player/mpv#13348]
### Important Information

Provide following Information:
- mpv version `master`
- macOS Version 13.6.3
- Source of the mpv binary or bundle  my own build using [mpv-build](https://github.com/mpv-player/mpv-build)
- If known which version of mpv introduced the problem 0341a6f
- Possible screenshot or video of visual glitches N/A

### Reproduction steps
- Using [mpv-build](https://github.com/mpv-player/mpv-build) run:
  -  ./use-mpv-custom 0341a6f
  -  ./update
  -  ./rebuild
- Copy libmpv.2.dylib into iina/deps/lib
- Build IINA with updated `libmpv`
- Run IINA from Xcode
- Start a video playing
- Pause playback
- Open a second video in another window
- That video plays, but with no audio

As soon as playback of the first video is paused the following two messages are emitted over and over again in the Xcode console:
```text
CompositeAudioConverter.cpp:1,236 Input data proc returned inconsistent 512 packets for 0 bytes; at 4 bytes per packet, that is actually 0 packets
CompositeAudioConverter.cpp:1,121 ProduceOutput: produced only 0 of 512 requested packets   Reinterleaver 0x6000011454a0
    Input:   2 ch,  48000 Hz, Float32, deinterleaved
    Output:  2 ch,  48000 Hz, Float32, interleaved
    Channel map: 0 1
```

With audio not playing if I change the audio device to a different set of speakers then the audio for the second video starts playing.

If I repeat the above steps, but switching to the commit before 0341a6f:
./use-mpv-custom 1888591

Audio works and the above errors are not seen in the console.

An IINA user tried upgrading `libmpv`, encountered this problem and reported it in https://github.com/iina/iina/issues/4789

### Expected behavior
Audio is heard through the speakers.

### Actual behavior
No audio is heard. Audio related errors are reported in the Xcode console.

### Log file
[mpv.log](https://github.com/mpv-player/mpv/files/14032318/mpv.log)

### Sample files
Any two videos with audio tracks should reproduce the problem.

**Comment by low-batt** [comment]
@handlerug Thoughts on this regression?

**Comment by low-batt** [comment]
Thank you very much for responding!

I pulled the latest master (fe0c181), applied the above patch, built `libmpv` and tested it using IINA. Audio worked fine. The messages seen in the Xcode console are no longer being emitted.

I read the discussion in PR #12643 about audio crackling on macOS. I have experienced crackling. I started paying attention to it and found I was only able to reproduce it when running under Xcode. It happens when I first open the IINA menu. This was before all of the latest changes. With the latest changes it still crackles, with or without the patch.

Since we are discussing audio on the Mac, another important problem to be aware of is issue #11617.

In testing with master I have encountered some sort of severe regression with playing an AV1 encoded video. I didn't find any open issues. Maybe my own build issue. I will investigate.
